profile/src/main/java/com/scottlogic/deg/profile/reader/ConstraintReader.java

```diff
@@ -207,58 +175,43 @@ private Constraint readGrammaticalConstraintDto(ConstraintDTO dto, ProfileFields
         }
     }
 
-    private int integer(Object value)
+    @Nullable
+    private Object readAnyType(Field field, Object value)
     {
-        return NumberUtils.coerceToBigDecimal(value).intValueExact();
+        switch (field.getType())
+        {
+            case DATETIME:
+                return HelixDateTime.create((String) value).getValue();
+            case NUMERIC:
+                return NumberUtils.coerceToBigDecimal(value);
+            default:
+                return value;
+        }
     }
 
-    private Pattern pattern(Object value)
+    private Pattern readPattern(Object value)
     {
-        return value instanceof Pattern ? (Pattern) value : Pattern.compile((String) value);
-    }
-
-    private OffsetDateTime parseDate(String value) {
-        if (value.toUpperCase().equals("NOW")) return NOW;
-        DateTimeFormatter formatter = new DateTimeFormatterBuilder()
-                .append(DateTimeFormatter.ofPattern("u-MM-dd'T'HH:mm:ss'.'SSS"))
-                .optionalStart()
-                .appendOffset("+HH", "Z")
-                .toFormatter()
-                .withResolverStyle(ResolverStyle.STRICT);
+        if (value instanceof Pattern) return (Pattern) value;
         try
         {
-            TemporalAccessor temporalAccessor = formatter.parse(value);
-
-            return temporalAccessor.isSupported(ChronoField.OFFSET_SECONDS)
-                    ? OffsetDateTime.from(temporalAccessor)
-                    : LocalDateTime.from(temporalAccessor).atOffset(ZoneOffset.UTC);
+            return Pattern.compile((String) value);
         }
-        catch (DateTimeParseException dtpe)
+        catch (IllegalArgumentException e)
         {
-            throw new InvalidProfileException(String.format(
-                    "Date string '%s' must be in ISO-8601 format: yyyy-MM-ddTHH:mm:ss.SSS[Z] between (inclusive) " +
-                            "0001-01-01T00:00:00.000Z and 9999-12-31T23:59:59.999Z",
-                    value
-            ));
+            throw new InvalidProfileException(e.getMessage());
         }
     }
 
-    private Granularity getOffsetUnit(FieldType type, String offsetUnit) {
+    private Granularity readGranularity(FieldType type, String offsetUnit) {
```

Comment:
I'm not sure this method name change is an improvement, can you consider reverting it if you're making changes to this PR anyway?

Reply:
The method doesn't get the offset unit, its already there in the parameter, instead it reads it and parses it into a granularity object. But it was probably superfluous to change it.

Reply:
the method name should definitely refer to offset, as there will always be another granularity in the class

getOffsetGranularity would be best imo